src/Caching/SqlServer/src/SqlServerCacheOptions.cs

```diff
@@ -43,6 +43,11 @@ public class SqlServerCacheOptions : IOptions<SqlServerCacheOptions>
         /// </summary>
         public TimeSpan DefaultSlidingExpiration { get; set; } = TimeSpan.FromMinutes(20);
 
+        /// <summary>
+        /// A function, if provided, should return an access token for authentication
+        /// </summary>
+        public Func<string> AccessTokenFunc { get; set; }
```

Comment:
Is there a reason this needs to be a function? Is the token expected to expire during while the application is running?

Reply:
Yes that's a possibility. The tokens issued by Azure for managed identity expire after 1 hour.